website/docs/interact-with-data.mdx

```diff
@@ -0,0 +1,287 @@
+          }
+        },
+        {
+          "node": {
+            "text": "My second post about ComposeDB!"
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+  
+
+You have options to retrieve specific records or last `n` indexed records as well. For example, to check the last 3 records, run the query below:
+
+  
+
+**Query:**
+
+  
+
+```graphql
+query{
+  postIndex(last: 3) {
+    edges {
+      node {
+        text
+      }
+    }
+  }
+}
+```
+  
+
+## Mutations
+There are two types of mutations you can perform on ComposeDB data: creating and updating records.
+  
+### Creating records
+Let’s say, you would like to create a post and add it to the graph. To do that, you will have to run a mutation as shown below and pass the actual text as a variable:
+
+  
+
+**Query:**
+
+  
+
+```graphql
+mutation CreateNewPost($i: CreatePostInput!){
+  createPost(input: $i){
+		document{
+			id
+      text
+    }
+  }
+}
+```
+
+  
+
+**Variables:**
+
+  
+
+```json
+{
+  "i": {
+    "content": {
+      "text": "A Post created using composites and GraphQL" 
+    }
+  }
+}
+```
+
+  
+
+The result of the query above will be a new document with a unique ID and the content you provided:
+
+  
+
+**Response**:
+
+  
+
+```json
+{
+  "data": {
+    "createPost": {
+      "document": {
+        "id": "kjzl6kcym7w8y9xlffqruh3v7ou1vn11t8203i6te2i3pliizt65ad3vdh5nl4l",
+        "text": "A Post created using composites and GraphQL"
+      }
+    }
+  }
+}
+```
+
+  
+> ✏️ ***Note:*** Stream IDs are unique. The “id” you will see in the response when performing the mutation above will be different.
+  
+
+### Updating records
+Now let’s say you want to edit the post you created in the previous step. To update it, you have to run the `UpdatePost` mutation and pass the post’s unique ID along with the updated content as variables.
+
+> 👀  You can find your post’s ID in the response after you ran the `CreateNewPost` mutation.
+  
+
+**Query:**
+
+```graphql
+mutation UpdatePost($i: UpdatePostInput!) {
+    updatePost(input: $i) {
+        document {
+            id
+            text
+        }
+    }
+}
+```
+
+  
+
+**Variables:**
+
+```json
+{{
+    "i": {
+          "id": "kjzl6kcym7w8y9xlffqruh3v7ou1vn11t8203i6te2i3pliizt65ad3vdh5nl4l",
+	        "content": {    
+            "text": "My best post!"
+        }
+    }
+}
+```
+
+This mutation will update the record with ID `kjzl6kcym7w8y9xlffqruh3v7ou1vn11t8203i6te2i3pliizt65ad3vdh5nl4l`.
+
+**Response:**
+```json
+{
+  "data": {
+    "updatePost": {
+      "document": {
+        "id": "kjzl6kcym7w8y9xlffqruh3v7ou1vn11t8203i6te2i3pliizt65ad3vdh5nl4l",
+        "text": "My best post!"
+      }
+    }
+  }
+}
+```
+
+## Authentication
+Although you can query records created by other accounts, you can only perform mutations on records controlled by your account. This guide did not require your authentication because you previously did that in the [Set up your environment](./set-up-your-environment) guide.
```

Comment:
```suggestion
Although you can query records created by other accounts, you can only perform mutations on records controlled by your account. This guide did not require your authentication because you previously did that in the [Set up your environment](./set-up-your-environment.mdx) guide.
```